app/state_machines/flood_risk_engine/enrollment_state_machine.rb

```diff
@@ -14,8 +14,54 @@ class EnrollmentStateMachine < FiniteMachine::Definition
     initial :check_location
 
     events do
-      event :go_forward, WorkFlow.for(:local_authority)
-      event :go_back, WorkFlow.for(:local_authority).invert
+      event :go_forward, WorkFlow.for(:start).merge(
+        if: -> { target.org_type.nil? }
+      )
+      event :go_back, WorkFlow.for(:start).invert.merge(
+        if: -> { target.org_type.nil? }
+      )
+
+      event :go_forward, WorkFlow.for(:local_authority).merge(
+        if: -> { target.org_type == "local_authority" }
+      )
+      event :go_back, WorkFlow.for(:local_authority).invert.merge(
+        if: -> { target.org_type == "local_authority" }
+      )
+
+      event :go_forward, WorkFlow.for(:limited_company).merge(
+        if: -> { target.org_type == "limited_company" }
+      )
+      event :go_back, WorkFlow.for(:limited_company).invert.merge(
+        if: -> { target.org_type == "limited_company" }
+      )
+
+      event :go_forward, WorkFlow.for(:limited_liability_partnership).merge(
+        if: -> { target.org_type == "limited_liability_partnership" }
+      )
+      event :go_back, WorkFlow.for(:limited_liability_partnership).invert.merge(
+        if: -> { target.org_type == "limited_liability_partnership" }
+      )
+
+      event :go_forward, WorkFlow.for(:individual).merge(
+        if: -> { target.org_type == "individual" }
+      )
+      event :go_back, WorkFlow.for(:individual).invert.merge(
+        if: -> { target.org_type == "individual" }
+      )
+
+      event :go_forward, WorkFlow.for(:partnership).merge(
+        if: -> { target.org_type == "partnership" }
+      )
+      event :go_back, WorkFlow.for(:partnership).invert.merge(
+        if: -> { target.org_type == "partnership" }
+      )
+
+      event :go_forward, WorkFlow.for(:other).merge(
+        if: -> { target.org_type == "other" }
+      )
+      event :go_back, WorkFlow.for(:other).invert.merge(
+        if: -> { target.org_type == "other" }
+      )
     end
   end
 end
```

Comment:
I have aboslutely no idea if this runs, but treat as a mad pseudo code idea for reducing noise a bit

```ruby
events do
  %i(partnership, other, individual).each do |organisation_type|
    is_current = -> { target.organisation_type == organisation_type.to_s }
    create_go_forward_event_for(organisation_type, is_current)
    create_go_back_event_for(organisation_type, is_current)
  end
end

def self.create_go_forward_event_for(organisation_type, org_type_is_current)
  steps = WorkFlow.for(organisation_type)
  event :go_forward, steps.merge(if: org_type_is_current))
end

def self.create_go_back_event_for(organisation_type, org_type_is_current)
  steps = WorkFlow.for(organisation_type).invert
  event :go_back, steps.merge(if: org_type_is_current)
end
```